File: restaurant_bar_app_server/models/producto.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const schema = new Schema({

    nombre: {
        type: String,
        required: true
    },
    precio: {
        type: Number,
        required: true
    },
    descripcion: {
        type: String,
        required: true
    },
    disponible: {
        type: Boolean,
        required: true,
        //default: true
    },
});

const model = mongoose.model('Producto', schema);
module.exports = model



// const { Schema, model } = require('mongoose');

// const ProductoSchema = Schema({

//     nombre: {
//         type: String,
//         required: true
//     },
//     precio: {
//         type: Number,
//         required: true
//     },
//     descripcion: {
//         type: String,
//         required: true
//     },
//     disponible: {
//         type: Boolean,
//         required: true,
//         //default: true
//     },

// });

// ProductoSchema.method('toJSON', function () {
//     const { __v, ...object } = this.toObject();
//     return object;
// })


// module.exports = model('Producto', ProductoSchema);

File: restaurant_bar_app/lib/models/producto.dart
class Producto {
    Producto({
        this.id,
        this.nombre,
        this.descripcion,
        this.precio,
        this.disponible,
    });

    String id;
    String nombre;
    String descripcion;
    double precio;
    bool disponible;

    factory Producto.fromJson(Map<String, dynamic> json) => Producto(
        id: json["_id"],
        nombre: json["nombre"],
        descripcion: json["descripcion"],
        precio: json["precio"].toDouble(),
        disponible: json["disponible"] == null ? null : json["disponible"],
    );

    Map<String, dynamic> toJson() => {
        "_id": id,
        "nombre": nombre,
        "descripcion": descripcion,
        "precio": precio,
        "disponible": disponible == null ? null : disponible,
    };
}